File: atcoder/abc395b_make_target.ml
let create_grid n =
  let grid = Array.make_matrix (n + 1) (n + 1) '#' in
  for i = 1 to n do
    let j = n + 1 - i in
    if i <= j then
      for k = i to j do
        for l = i to j do
          grid.(k).(l) <- if i mod 2 = 1 then '#' else '.'
        done
      done
  done;
  for i = 1 to n do
    print_endline (String.sub (Array.fold_left (^) "" (Array.map (String.make 1) grid.(i))) 1 n)
  done

let () =
  let n = read_int () in
  create_grid n

File: atcoder/abc395b_make_target.nim
import strutils, sequtils

proc createGrid(n: int) =
  var grid = newSeqWith(n + 1, newSeqWith(n + 1, '#'))
  
  for i in 1..n:
    let j = n + 1 - i
    if i <= j:
      for k in i..j:
        for l in i..j:
          if i mod 2 == 1:
            grid[k][l] = '#'
          else:
            grid[k][l] = '.'
  
  for i in 1..n:
    echo grid[i][1..n].join("")

when isMainModule:
  let n = parseInt(readLine(stdin))
  createGrid(n)
